src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -797,44 +800,76 @@ private function registerValidationConfiguration(array $config, ContainerBuilder
         }
     }
 
-    private function getValidatorMappingFiles(ContainerBuilder $container)
+    /**
+     * Get all validation files.
+     *
+     * @param ContainerBuilder $container A ContainerBuilder instance
+     * @param array            $files     An array reference of validation files
+     */
+    private function getValidatorMappingFiles(ContainerBuilder $container, array &$files)
     {
-        $files = array(array(), array());
-
         if (interface_exists('Symfony\Component\Form\FormInterface')) {
             $reflClass = new \ReflectionClass('Symfony\Component\Form\FormInterface');
-            $files[0][] = dirname($reflClass->getFileName()).'/Resources/config/validation.xml';
-            $container->addResource(new FileResource($files[0][0]));
+            $files['xml'][] = $file = dirname($reflClass->getFileName()).'/Resources/config/validation.xml';
+            $container->addResource(new FileResource($file));
         }
 
         $bundles = $container->getParameter('kernel.bundles');
         foreach ($bundles as $bundle) {
             $reflection = new \ReflectionClass($bundle);
             $dirname = dirname($reflection->getFileName());
 
-            if (is_file($file = $dirname.'/Resources/config/validation.xml')) {
-                $files[0][] = realpath($file);
+            if (is_file($file = $dirname.'/Resources/config/validation.yml')) {
+                $files['yml'][] = realpath($file);
                 $container->addResource(new FileResource($file));
             }
 
-            if (is_file($file = $dirname.'/Resources/config/validation.yml')) {
-                $files[1][] = realpath($file);
+            if (is_file($file = $dirname.'/Resources/config/validation.xml')) {
+                $files['xml'][] = realpath($file);
                 $container->addResource(new FileResource($file));
             }
 
             if (is_dir($dir = $dirname.'/Resources/config/validation')) {
-                foreach (Finder::create()->files()->in($dir)->name('*.xml') as $file) {
-                    $files[0][] = $file->getRealpath();
-                }
-                foreach (Finder::create()->files()->in($dir)->name('*.yml') as $file) {
-                    $files[1][] = $file->getRealpath();
-                }
-
+                $this->getValidatorMappingFilesFromDir($dir, $files);
                 $container->addResource(new DirectoryResource($dir));
             }
         }
+    }
 
-        return $files;
+    /**
+     * Searches for validation files in directory.
+     *
+     * @param string $dir   The directory that contains the validation files
+     * @param array  $files An array reference of validation files
+     */
+    private function getValidatorMappingFilesFromDir($dir, array &$files)
+    {
+        foreach (Finder::create()->files()->in($dir)->name('/\.(xml|ya?ml)$/') as $file) {
+            $files[$file->getExtension()][] = $file->getRealpath();
```

Comment:
Can't you merge directly `yml` and `yaml` files here to avoid the `array_merge` later?

```php
$extension = $file->getExtension();
$files['yml' === $extension ? 'yaml' : $extension][] = $file->getRealpath();
```